lib/client/api_test.go

```diff
@@ -500,3 +504,108 @@ func TestApplyProxySettings(t *testing.T) {
 		})
 	}
 }
+
+type mockAgent struct {
+	// Agent is embedded to avoid redeclaring all interface methods.
+	// Only the Signers method is implemented by testAgent.
+	agent.Agent
+	ValidPrincipals []string
+}
+
+type mockSigner struct {
+	ValidPrincipals []string
+}
+
+func (s *mockSigner) PublicKey() ssh.PublicKey {
+	return &ssh.Certificate{
+		ValidPrincipals: s.ValidPrincipals,
+	}
+}
+
+func (s *mockSigner) Sign(rand io.Reader, b []byte) (*ssh.Signature, error) {
+	return nil, trace.Errorf("mockSigner does not implement Sign")
+}
+
+// Signers implements agent.Agent.Signers.
+func (m *mockAgent) Signers() ([]ssh.Signer, error) {
+	return []ssh.Signer{&mockSigner{ValidPrincipals: m.ValidPrincipals}}, nil
+}
+
+func TestNewClient_UseKeyPrincipals(t *testing.T) {
+	cfg := &Config{
+		Username:         "xyz",
+		HostLogin:        "xyz",
+		WebProxyAddr:     "localhost",
+		SkipLocalAuth:    true,
+		UseKeyPrincipals: true, // causes VALID to be returned, as key was used
+		Agent:            &mockAgent{ValidPrincipals: []string{"VALID"}},
+		AuthMethods:      []ssh.AuthMethod{ssh.Password("xyz") /* placeholder authmethod */},
+	}
+	client, err := NewClient(cfg)
+	require.NoError(t, err)
+	require.Equal(t, "VALID", client.getProxySSHPrincipal(), "ProxySSHPrincipal mismatch")
+
+	cfg.UseKeyPrincipals = false // causes xyz to be returned as key was not used
+
+	client, err = NewClient(cfg)
+	require.NoError(t, err)
+	require.Equal(t, "xyz", client.getProxySSHPrincipal(), "ProxySSHPrincipal mismatch")
+}
+
+func TestParseSearchKeywords(t *testing.T) {
```

Comment:
This isn't really part of the backport, is it?

Reply:
Ah sorry, mistake with a merge

Reply:
No worries, same thing happened with me yesterday.